frontend/server/src/Controllers/Tag.php

```diff
@@ -65,4 +65,42 @@ public static function getPublicTags() {
             'problemTag'
         );
     }
+
+    public static function getFrequentTagsByLevel(
+        string $problemLevel
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::TAGS_LIST,
+            'frequent',
+            fn () => \OmegaUp\DAO\Tags::getFrequentsByLevel(
+                $problemLevel
+            ),
+            APC_USER_CACHE_SESSION_TIMEOUT
+        );
+    }
+
+    /**
+     * Return most frequent public tags of a certain level
+     *
+     * @return list<array{name: string}>
+     */
+    public static function apiFrequentTags(\OmegaUp\Request $r): array {
+        $param = '';
+        if (is_string($r['level'])) {
+            $param = $r['level'];
+        } else {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterEmpty',
+                'level'
+            );
+        }
+
+        $response = [];
+        foreach (self::getFrequentTagsByLevel($param) as $tag) {
+            $response[] = [
+                'name' => strval($tag->name),
+            ];
+        }
+        return $response;
```

Comment:
Con el cambio que te propongo en el DAO, aquí ya podrías regresar directamente el valor de la función:
```suggestion
        return self::getFrequentTagsByLevel($param);
```